buildSrc/src/main/kotlin/org/jetbrains/exposed/gradle/Publishing.kt

```diff
@@ -65,31 +65,52 @@ fun Project.java(configure: JavaPluginExtension.() -> Unit) {
     extensions.configure("java", configure)
 }
 
-fun Project.configurePublishing() {
-    apply(plugin = "java-library")
-    apply(plugin = "maven-publish")
-    apply(plugin = "signing")
-
-    java {
-        withJavadocJar()
-        withSourcesJar()
+fun Project.configurePublishing(isVersionCatalog: Boolean = false) {
```

Comment:
@jetbrains-junie fix

Split this into a separate function instead of using a boolean flag.

Re-exclude exposed-version-catalog from allprojects `configurePublishing` in `build.gradle.kts`.

Call the new `configureVersionCatalogPublishing` from the version catalog build file.